core/server/src/main/java/alluxio/master/file/FileSystemMaster.java

```diff
@@ -658,19 +661,19 @@ public StartupConsistencyCheck getStartupConsistencyCheck() {
    * @throws AccessControlException if permission checking fails
    */
   public long getFileId(AlluxioURI path) throws AccessControlException {
-    long flushCounter = AsyncJournalWriter.INVALID_FLUSH_COUNTER;
+    Mutable<Long> flushCounter = new MutableObject<>(AsyncJournalWriter.INVALID_FLUSH_COUNTER);
```

Comment:
the closable ordering could be taken care of by creating the resources in the right order (flush counter first)

what is more challenging is invoking `waitForJournalFlush` in the `close` method ... the reason for that is that `waitForJournalFlush` is a protected method of `AbstractMaster` and the method uses `mAsyncJournalWriter`, which is a private member of `AbstractMaster` ... I see a number of solutions here: 1) put FlushCounter into the same package as AbstractMaster so that it has access to waitForJournalFlush, 2) relax the access of waitForJournalFlush, 3) add a getter for mAsyncJournalWriter, 4) introduce `FlushCounter` as a nested class for AbstractMaster ... I don't like any of these options, but if I had to choose, I would choose either 1) or 4) ... what are your thoughts?